### PR TITLE
Differentiable classical force field support

### PR DESCRIPTION
**Pull request by WangXinyan940**
We propose to add the following features to DMFF:

- 'LennardJonesForce'
- 'CoulombPMEForce'
- 'CoulNoCutoffForce'
- 'CoulReactionFieldForce'
- 'HarmonicBondJaxForce'
- 'HarmonicAngleJaxForce'
- 'PeriodicTorsionJaxForce'

The following generators are added to support related cards in OpenMM type XML forcefield.
- 'HarmonicBondJaxGenerator'
- 'HarmonicAngleJaxGenerator'
- 'PeriodicTorsionJaxGenerator'
- 'NonbondJaxGenerator'

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -470,6 +470,14 @@ StyleCopReport.xml
...
 *.meta
 *.iobj
+<<<<<<< HEAD
+*.ipdb
+=======
+<<<<<<< HEAD
+*.ipdb
+=======
+>>>>>>> 27b8bde1c4744db2f9ce5623fcebd9f764693073
+>>>>>>> 70756cd80617652580988e6ef6b04545df86f7ca
```
Comment (Ericwang6) [inline .gitignore:480]: This should be solved...

File: dmff/classical/inter.py [hunk 2]
```diff
@@ -1,5 +1,335 @@
+import sys
+
+sys.path.append("/home/lijichen/work/DMFF")
```
Comment (Ericwang6) [inline dmff/classical/inter.py:3]: Here should be solved

**Review by Ericwang6 — CHANGES_REQUESTED** [review]

**Review by KuangYu — CHANGES_REQUESTED** [review]
See comments in code.

File: dmff/classical/inter.py [hunk 3]
```diff
@@ -1,5 +1,335 @@
...
+
+            if self.isSwitch:
+
+                x = (dr_norm - self.r_switch) / (self.r_cut - self.r_switch)
+                S = 1 - 6. * x ** 5 + 15. * x ** 4 - 10. * x ** 3
+                jnp.where(dr_norm > self.r_switch, E, E * S)
+
+            return E
+
+        def get_energy(positions, box, pairs, epsilon, sigma, epsfix, sigfix):
```
[the remark this answers is not in the document]
Comment (WangXinyan940) [reply dmff/classical/inter.py:63]: I'd like to input position/box in nm & kJ/mol in API, and then transfer to angstrom when creating potential function like this:

class ADMPDispGenerator:
    def createForce():
        def potential_fn(positions, box, pairs, params):
            position_ang = positions * 10.0 
            box_ang = box * 10.0
            E_sr = pot_fn_sr(
                position_ang, box_ang, pairs, mScales, a_list, b_list, q_list, c_list[0]
            )
            E_lr = pot_fn_lr(position_ang, box_ang, pairs, c_list.T, mScales)
            return E_sr - E_lr

Then we don't need to touch parameter preparing process.

File: dmff/classical/inter.py [hunk 4]
```diff
@@ -1,5 +1,335 @@
...
+            )
+            excl_map0 = self.map_prm[self.map_exclusion[:, 0]]
+            excl_map1 = self.map_prm[self.map_exclusion[:, 1]]
+            eps_excl = eps_mat[excl_map0, excl_map1]
+            sig_excl = sig_mat[excl_map0, excl_map1]
+
+            E_excl = get_LJ_energy(dr_excl_vec, sig_excl, eps_excl, box)
+            E_excl = self.scale_exclusion * E_excl
+
+            return jnp.sum(E_inter) - jnp.sum(E_excl)
```
[the remark this answers is not in the document]
Comment (WangXinyan940) [reply dmff/classical/inter.py:98]: OK. Making change now.

File: dmff/classical/inter.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (KuangYu) [reply dmff/classical/inter.py:63]: This is good. So let's make it a rule:

**Pull request merged**